chainservice/builder.go

```diff
@@ -430,7 +431,10 @@ func (builder *Builder) buildBlockSyncer() error {
 			return nil
 		},
 		p2pAgent.ConnectedPeers,
-		p2pAgent.UnicastOutbound,
+		// TODO: WithBatch() will be added after the hardfork
+		func(ctx context.Context, peer peer.AddrInfo, msg proto.Message) error {
```

Comment:
can we avoid change the func signature?
a single message or batch message both fit `proto.Message` right? we can identify the message type inside the actual unicast func?

Reply:
yes, it can work. But it will add some conditions into the func

Reply:
thx for the discussion last night, i still think define 2 new `MsgType` for actions and blocks would be a better option:
1. design-wise, the p2p message is self-contained/explanatory, `MsgType` tells what type of proto data (single or concatenated) was in the message body
2. coding-wise, we don't need to modify the API interface here
3. structure-wise, it also reduce complexity/coupling -- p2p agent does not need to know the existence of a `batch mode` at API level, it just receive a `proto.Message` msg, and handles it accordingly (you know what to do from `msg.MsgType`)